source/Nevermore/Advanced/ReaderStrategies/Documents/DocumentReaderExpressionBuilder.cs

```diff
@@ -281,9 +281,6 @@ void AssertValidColumnOrdering()
             if (expectsType && typeIndex < 0)
                 throw Fail($"When querying the document '{name}', the '{map.TypeResolutionColumn.ColumnName}' column must always be included. Change the select clause to include the '{map.TypeResolutionColumn.ColumnName}' column.");
 
-            if (typeIndex >= 0 && expectsJson && typeIndex > jsonIndex)
-                throw Fail($"When querying the document '{name}', the 'Type' column must always appear before the 'JSON' column. Change the order in the SELECT clause, or if selecting '*', change the order of columns in the table.");
```

Comment:
This still feels like a reasonable safety net, is it still necessary to remove this? Shouldn't we now be getting the columns ordered in such a way to avoid this perf issue?

Reply:
Good call - we should be keeping this for a case where someone might write their own query and not select in the right order.